### ISSUE TITLE
SPRT LLR calculation

### ISSUE THREAD

**Issue by sethtroisi** [sethtroisi/cloudygo#34]
I'm writing SPRT calculator and I'm confused by the sprt math.

I wrote up the three methods I know in this [gist](https://gist.github.com/sethtroisi/b1be942f4b918a20dd5388b6d06fbbdc)
LLF = my optimized version
wiki = my take on [likelihood-ratio_test](https://en.wikipedia.org/wiki/Likelihood-ratio_test#Simple_hypotheses)
LLR = the code from [leela-zero-server](https://github.com/gcp/leela-zero-server/blob/master/classes/utilities.js#L178)

Other links:
https://www.chessprogramming.org/Match_Statistics#SPRT
^ uses similar math to this repo but with draws
^ draws it's justification from http://hardy.uhasselt.be/Toga/GSPRT_approximation.pdf

https://nowak.ece.wisc.edu/ece830/ece830_fall11_lecture9.pdf
^ maybe I'm misunderstanding p0(x) and p1(x)

**Comment by sethtroisi** [comment]
I think maybe it's because I'm testing likelihood that θ = θ1 not θ > θ1

**Comment by roy7** [comment]
Yeah we don't care if it's exactly 35 ELO better, only that is 35 or more better. On the lower end test we'd tweaked it to test at 0 since if it's under 0 we don't care if it's -5 or -75, so why waste time figuring out if it's better or worse than -35.

**Comment by roy7** [comment]
Past discussions that might help (or not).

https://github.com/gcp/leela-zero/issues/378
https://github.com/gcp/leela-zero/issues/543
https://github.com/gcp/leela-zero/issues/545

**Comment by sethtroisi** [comment]
The author michel van den bergh has a bunch of comments on github and talkchess
http://talkchess.com/forum3/viewtopic.php?f=7&t=57465&start=20#p737797
